### DISCUSSION TITLE
Choosing between the CSI for K8S and the SeaweedFS operator,

### DISCUSSION THREAD

**Discussion by timothywcrane** [seaweedfs/seaweedfs#2344] [Q&A] [answered]
* do I lose K8S master node disk use capability when using SeaweedFS Operator (containers only on non-masters)?

* or do I just lose (temp I hope ;)) the CSI label for more speed/reliab - with same space capacity (all 4 nodes complete surface in a 4 node K8S cluster)?

* Will  I need the CSI to claim 4 all disks' capacity for SeaweedFS?

* * Am I missing something?

**Comment by chrislusf** [comment, marked as answer]
* no. Operator costs no disk space.
* yes
* Not necessary. CSI is basically a “weed mount”, which uses FUSE. There are other access methods: S3, HTTP, WebDAV, Hadoop, etc.

**Reply by timothywcrane** [reply]
Yippee Kiyaay! f^5ing A... thanks. Monday morning and good news.
